backends/ebpf/ebpftargets.py

```diff
@@ -207,120 +208,141 @@ def check_outputs(self):
+                      " bpf da obj %s section prog "
+                      "verbose" % (port_name, self.template + ".o"))
+        result = bridge.ns_proc_write(proc, cmd_ingress)
         if result != SUCCESS:
             return result
-        args = ["tc"]
-        # Launch tc to load the ebpf object to the specified interface
-        args.extend(["filter", "add", "dev", ifname, "ingress",
-                     "bpf", "da", "obj", self.template + ".o",
-                     "sec", "prog", "verb"])
-        # The bpf program to attach to the interface
-        errmsg = "Failed to load the eBPF byte code using tc:"
-        return run_timeout(self.options, args, TIMEOUT, self.outputs, errmsg)
+        return bridge.ns_proc_append(proc, cmd_egress)
+
+    def _run_in_namespace(self, bridge):
+        # Open a process in the new namespace
+        proc = bridge.ns_proc_open()
+        if not proc:
+            return FAILURE
+        # Get the command to load eBPF code to all the attached ports
+        if len(bridge.br_ports) > 0:
+            for port in bridge.br_ports:
+                result = self._load_tc_cmd(bridge, proc, port)
+                bridge.ns_proc_append(proc, "")
+        else:
+            # No ports attached (no pcap files), load to bridge instead
+            result = self._load_tc_cmd(bridge, proc, bridge.br_name)
+            bridge.ns_proc_append(proc, "")
 
-    def create_filter(self):
-        require_root(self.outputs)
-        # Create interface with unique id (allows parallel tests)
-        # TODO: Create one custom interface per pcap file we parse
-        ifname = str(os.getpid())
-        self._create_bridge(ifname)
-        # Use clang to compile the generated C code to a LLVM IR
-        result = self._compile_ebpf(self.ebpfdir)
         if result != SUCCESS:
-            self._remove_bridge(ifname)
             return result
-        result = self._load_ebpf(self.ebpfdir, ifname)
+        # Check if eBPF maps have actually been created
+        result = bridge.ns_proc_write(proc,
+                                      "ls -1 /sys/fs/bpf/tc/globals")
         if result != SUCCESS:
-            self._remove_bridge(ifname)
-        return result
+            return result
+        # Finally, append the actual runtime command to the process
+        result = bridge.ns_proc_append(proc, self._get_run_cmd())
+        if result != SUCCESS:
+            return result
+        # Execute the command queue and close the process, retrieve result
+        return bridge.ns_proc_close(proc)
 
-    def check_outputs(self):
-        # Not implemented yet, just pass the test
-        ifname = str(os.getpid())
-        self._remove_bridge(ifname)
-        return SUCCESS
+    def run(self):
+        # Root is necessary to load ebpf into the kernel
+        require_root(self.outputs)
+        result = self._create_runtime()
+        if result != SUCCESS:
+            return result
```

Comment:
I hate these "if result != SUCCESS:  return result" blocks, they take up a lot of space. Is there a more succinct way to implement this in Python?

Reply:
Don't worry about this.